week04/index.html

```diff
@@ -0,0 +1,163 @@
+            </div>
+            <div class="text">버튼을 눌러 시작하세요!</div>
+        </div>
+        
+        <div class="middle">
+            <div id="human_select"><img id="img" src="./img/paper.png"/></div>
+            <div id="robot_select"><img id="robotimg" src="./img/paper.png"/></div>
+        </div>
+        <div class="bottom">
+            <button onclick="scissor_button()" id="img_toScissor">가위</button>
+            <button onclick="rock_button()" id="img_toRrock">바위</button>
+            <button onclick="paper_button()" id="img_toPaper">보</button>
+        </div> 
+        <button id="reset">초기화</button>
+    </div>
+
+    <script>
+        let humanScore  = document.getElementById("humanscore");
+        let computerScore = document.getElementById("computerscore");
+        const resetbutton = document.getElementById("reset");
+        const text = document.getElementsByClassName("text");
+        let human = 0;
+        let computer = 0;
+        resetbutton.onclick = ()=>{
+            computerScore.innerText = "0";
+            humanScore.innerText = "0";
+            //승리 출력
+            const text = document.getElementsByClassName("text");
+                for (let i = 0; i < text.length; i++) {
+                    text[i].innerText = "버튼을 눌러 시작하세요!";
+                }
+        };
+
+        function rock_button() {
+            
+            document.getElementById("img").src = "./img/rock.png";
+            const human = 2;
+            const computer = Math.floor(Math.random() * 3) + 1;
+            console.log(computer);
+            /*컴퓨터와 비교 후 점수 올리고 내리고*/
+            if(computer === human){
+                document.getElementById("robotimg").src = "./img/rock.png";
+                //비겼을 때
+                const text = document.getElementsByClassName("text");
+                for (let i = 0; i < text.length; i++) {
+                    text[i].innerText = "비김";
+                }
+            }
+            else if(human > computer){
+                const humanScore = parseInt(humanscore.innerText);
+                humanscore.innerText = humanScore+1;
+                console.log("승리");
+                //승리 출력
+                const text = document.getElementsByClassName("text");
+                for (let i = 0; i < text.length; i++) {
+                    text[i].innerText = "승";
+                }
+                document.getElementById("robotimg").src = "./img/scissor.png";
+            }
+            else if(human < computer){
+                //패배 출력
+                const text = document.getElementsByClassName("text");
+                for (let i = 0; i < text.length; i++) {
+                    text[i].innerText = "패";
+                }
+                const computerScore = parseInt(computerscore.innerText);
+                computerscore.innerText = computerScore+1;
+                document.getElementById("robotimg").src = "./img/paper.png";
+            }
+        };
```

Comment:
로직이 정상적이지 않습니다. 승리, 무승부, 패배의 상황을 먼저 결정한 후 computer의 패를 결정하는 것은 가위바위보 게임이 아닙니다. 이 로직은 실제 가위바위보 게임의 경우의 수와도 맞지 않습니다. 각자의 패를 결정한 후 비교하여 승패를 가르는 것이 가위바위보 게임입니다. 코드를 참고하여 로직을 다시 만들어보세요.

```suggestion
        function rock_button() {
        //가위=1, 바위=2, 보=3
        
           const human_select = 2;
           const computer_select = Math.floor(Math.random() * 3) + 1;
           //이미지 표시 코드
           
           if(computer_select === human_select){ //무승부 시
              text.innerText = "비김";
           } else if((human_select==1 && computer_select==3)||(human_select==2 && computer_select==1)||(human_select==3 && computer_select==2)) { //승리 시
              human_score++;
              text.innerText = "승";
           } else if((human_select==1 && computer_select==2)||(human_select==2 && computer_select==3)||(human_select==3 && computer_select==1)) { //패배 시
              computer_score++;
              text.innerText = "패";
           }
           
           score_text.innerText = `${human_score} : ${computer_score}`;
       }
```